bika/lims/subscribers/setup.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2019 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import api
+from bika.lims import permissions
+
+
+def ObjectModifiedEventHandler(instance, event):
+    """Actions to be taken when Setup object is modified
+    """
+    update_worksheet_manage_permissions(instance)
```

Comment:
This code fires on every setup object modification and reindexes all worksheets, which slows down the setup for sites with many WSs.

It would be better to remember the last setting of the `RestrictWorksheetManagement` in a hidden field, e.g. `_RestrictWorksheetManagement` and only update the permissions if `RestrictWorksheetManagement != _RestrictWorksheetManagement`

Reply:
Keep in mind that worksheet objects are not reindexed, only worksheets folder. In my opinicion, since only `worksheets` folder is reindexed, I think adding a new field as a discriminator is not worth.